code/pagetypes/ForumHolder.php

```diff
@@ -252,11 +242,24 @@ function getNumTopics() {
 	 * @return int Returns the number of distinct authors
 	 */
 	public function getNumAuthors() {
-		return DB::query("
-			SELECT COUNT(DISTINCT \"Post\".\"AuthorID\") 
-			FROM \"Post\" 
-			JOIN \"" . ForumHolder::baseForumTable() . "\" AS \"ForumPage\" ON \"Post\".\"ForumID\"=\"ForumPage\".\"ID\"
-			AND \"ForumPage\".\"ParentID\" = '" . $this->ID . "'")->value(); 		
+		return $this->getForumStatistics()['Authors'];
+	}
+
+	/**
+	 * Tally the number of posts, threads and authors from the Forum of this ForumHolder
+	 *
+	 * @return array Returns an key => value pair of the stats for the forum.
+	 */
+	public function getForumStatistics() {
```

Comment:
I would recommend going back to a SQL only based approach. It can be awkward to have to re-implement business logic in sql, but you get the benefit of sql query caches, and not having to run tasks in o(n) for each request, (or is this o(n^2)? ). This is going to be a huge performance bottleneck.